lib/client/jsx/components/search/search.jsx

```diff
@@ -2,138 +2,216 @@ import Pager from '../pager';
 import React, {Component} from 'react';
 import {connect} from 'react-redux';
 
+import {css} from '@emotion/core';
+import GridLoader from 'react-spinners/GridLoader';
+
 import SelectInput from '../inputs/select_input';
-import { selectModelNames } from '../../selectors/magma';
-import { requestTSV, requestModels, requestDocuments } from '../../actions/magma_actions';
-import { selectSearchCache } from '../../selectors/search_cache';
-import { cacheSearchPage, setSearchPageSize, setSearchPage, emptySearchCache } from '../../actions/search_actions';
+import {selectModelNames} from '../../selectors/magma';
+import {
+  requestTSV,
+  requestModels,
+  requestDocuments
+} from '../../actions/magma_actions';
+import {
+  selectSearchCache,
+  selectSearchAttributes
+} from '../../selectors/search_cache';
+import {
+  cacheSearchPage,
+  setSearchPageSize,
+  setSearchPage,
+  emptySearchCache,
+  setSearchAttributes
+} from '../../actions/search_actions';
 
 import ModelViewer from '../model_viewer';
 
+const spinnerCss = css`
+  display: block;
+  margin: 2rem auto;
+`;
+
 class Search extends Component {
   constructor(props) {
-    super(props)
-    this.state = { page_size: 10 }
+    super(props);
+    this.state = {page_size: 10, loading: false};
   }
 
-  getPage(page, newSearch=false) {
+  getPage = (page, newSearch = false) => {
+    let {attributes} = this.props;
+
     page = page + 1;
+
     if (!this.pageCached(page) || newSearch) {
+      this.setState({loading: true});
       this.props.requestDocuments({
         model_name: this.state.selected_model,
         record_names: 'all',
-        attribute_names: 'all',
+        attribute_names: attributes,
         filter: this.state.current_filter,
         page: page,
         page_size: this.state.page_size,
         collapse_tables: true,
         exchange_name: `request-${this.state.selected_model}`,
-        success: this.makePageCache.bind(this, page, 
-          newSearch ? this.state.page_size : null)
-      })
+        success: this.makePageCache.bind(
+          this,
+          page,
+          newSearch ? this.state.page_size : null
+        ),
+        error: this.handleError
+      });
     }
-    this.props.setSearchPage(page)
-  }
+    this.props.setSearchPage(page);
+  };
 
   pageCached(page) {
-    let { cache } = this.props;
-    return cache.isCached(page.toString())
+    let {cache} = this.props;
+    return cache.isCached(page.toString());
   }
 
   componentDidMount() {
-    this.props.requestModels()
+    this.props.requestModels();
     this.props.emptySearchCache();
+
+    // @Zach -- This sets the search attributes to the default.
+    // When we have UI widgets to control this, we
+    //   will need to tie the setSearchAttributes() action
+    //   into those widgets and remove this.
+    // setSearchAttributes() expects 'all' or a list of attribute names.
+    this.props.setSearchAttributes('all');
```

Comment:
Sets the default value to `"all"` `attributes` when the Search component mounts. I guess we wouldn't really remove this call like the comment says, but would rather call `setSearchAttributes()` again from the new widgets.